interfaces/cython/cantera/test/test_composite.py

```diff
@@ -119,6 +138,90 @@ def check(a, b):
                     raise TypeError(msg) from inst
 
 
+class TestSolutionArrayIO(utilities.CanteraTest):
+    """ Test SolutionArray file IO """
+    @classmethod
+    def setUpClass(cls):
+        utilities.CanteraTest.setUpClass()
+        cls.gas = ct.Solution('h2o2.yaml')
+
+    def test_write_csv(self):
+        states = ct.SolutionArray(self.gas, 7)
+        states.TPX = np.linspace(300, 1000, 7), 2e5, 'H2:0.5, O2:0.4'
+        states.equilibrate('HP')
+
+        outfile = pjoin(self.test_work_dir, 'solutionarray.csv')
+        states.write_csv(outfile)
+
+        data = np.genfromtxt(outfile, names=True, delimiter=',')
+        self.assertEqual(len(data), 7)
+        self.assertEqual(len(data.dtype), self.gas.n_species + 2)
+        self.assertIn('Y_H2', data.dtype.fields)
+
+        b = ct.SolutionArray(self.gas)
+        b.read_csv(outfile)
+        self.assertTrue(np.allclose(states.T, b.T))
+        self.assertTrue(np.allclose(states.P, b.P))
+        self.assertTrue(np.allclose(states.X, b.X))
+
+    def test_to_pandas(self):
+        if isinstance(_pandas, ImportError):
+            return
```

Comment:
Rather than giving the impression that this test passes when `pandas` is not installed, I would suggest conditionally skipping the test instead, using the `skipIf` decorator. I think it should be something like:
```suggestion
    @unittest.skipIf(isinstance(_pandas, ImportError), "pandas is not installed")
    def test_to_pandas(self):
```
And likewise for the tests requiring h5py.